### PR TITLE
Casper txn id support on ledger desktop and mobile

### PR DESCRIPTION
**Pull request by lawRathod**
* Adds txnId to transaction description 
* Adds txnId field to send flow

### CODE REVIEW

File: libs/ledger-live-common/src/families/casper/bridge/account.ts [hunk 1]
```diff
@@ -104,6 +110,9 @@ const getTransactionStatus = async (
   else if (!validateAddress(address).isValid)
     errors.sender = new InvalidAddress();
 
+  if (t.transferId && !validateTransferId(t.transferId).isValid)
```
Comment (emmanuelm41) [inline libs/ledger-live-common/src/families/casper/bridge/account.ts]: Have we added a test case for this on the integration test suite?

File: libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts [hunk 2]
```diff
@@ -37,7 +44,9 @@ export function mapTxToOps(accountId: string, addressHash: string) {
         senders: [fromAccount],
         recipients: [toAccount],
         date,
-        extra: {},
+        extra: {
+          "Transfer ID": transferId,
```
Comment (emmanuelm41) [inline libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: I am not really convinced with this. Why are we using object keys with spaces in between?

File: libs/ledger-live-common/src/families/casper/bridge/account.ts [hunk 3]
```diff
@@ -104,6 +110,9 @@ const getTransactionStatus = async (
...
+    errors.sender = new Error("Invalid TransferID");
```
Comment (emmanuelm41) [inline libs/ledger-live-common/src/families/casper/bridge/account.ts]: Are we sure this error will be displayed correctly on frontend?

File: libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts (hunk 2 above)
Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: When I used this, the field automatically got displayed in the Operation details. So if it's "transferID" it doesn't look nice.

File: libs/ledger-live-common/src/families/casper/bridge/account.ts (hunk 3 above)
Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/account.ts]: Well, on the frontend, entering anything but numbers is non permissible. So this error will never be fired on the frontend.

File: libs/ledger-live-common/src/families/casper/bridge/account.ts (hunk 1 above)
Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/account.ts]: Missed it, can add...

Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/account.ts]: Done

File: libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts (hunk 2 above)
Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: What is mean is, when a key-value is added to extra I think it automatically appear on the operationInformation. Therefore the keys need to be named the way we want them to be displayed in the UI

File: libs/ledger-live-common/src/families/casper/bridge/account.ts (hunk 3 above)
Comment (emmanuelm41) [reply libs/ledger-live-common/src/families/casper/bridge/account.ts]: Great!

File: libs/ledger-live-common/src/families/casper/bridge/account.ts (hunk 1 above)
Comment (emmanuelm41) [reply libs/ledger-live-common/src/families/casper/bridge/account.ts]: Perfect!

File: libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts (hunk 2 above)
Comment (emmanuelm41) [reply libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: Yes, I remember from other integrations now that you mentioned it.

Comment (lawRathod) [reply libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: Done, added operationDetails.tsx. TransferId on mobile now pending. I am working on it.

Comment (emmanuelm41) [reply libs/ledger-live-common/src/families/casper/bridge/utils/txn.ts]: Great!

**Pull request merged**